packages/compiler/core/types.ts

```diff
@@ -258,49 +258,46 @@ export interface TemplateParameterType extends BaseType {
   node: TemplateParameterDeclarationNode;
 }
 
-export interface ProjectionSymbol {
-  kind: "projection";
-  name: string;
-  node: ProjectionStatementNode;
-  byKind: Map<string, { to?: ProjectionNode; from?: ProjectionNode }>;
-  byId: Map<TypeReferenceNode, { to?: ProjectionNode; from?: ProjectionNode }>;
-  id?: number;
-  symbolSource?: ProjectionSymbol;
-}
+export interface Sym {
+  flags: SymbolFlags;
```

Comment:
Another observation about state prior to this PR (so not blocking this PR). If symbols are in nodes and symbols are mutably typed, then nodes are still mutably typed. IIRC, TS hides symbol from outside consumers with `/** @internal */? Should we do likewise maybe?

Reply:
No problem with making symbols internal. I would like symbols to be immutable but I suspect that decorators adding/removing things during compilation means that symbols may change over the lifetime of the program. For performance reasons it might be worth splitting static symbols from late-bound symbols (so we can reuse the result of bidning) which might make it possible to treat our present symbols as read-only.

Reply:
On second thought I think the symbol flags should be part of our API surface so @internal doesn't feel correct to me.